### PR TITLE
LPS-159215 Include pathContext when building the URLs from the UriInfo in Vulcan

### PR DESCRIPTION
**Pull request by ccorreagg**
This PR contains the code to include the pathContext when building the URLs in the UriInfoUtil in Vulcan that is used to build certain URLs in the REST APIs.

---

### **Before the PR:**

If there is a pathContext defined in the portal (see the LPS-159215 in Jira to find more details about how to configure/reproduce it) when the API Explorer is accessed, an error occurs:

image

As the URL is not the correct one, the pathContext is not being included:

`Not Found http://localhost/o/headless-delivery/v1.0/openapi.json`

Furthermore, the URLs built in `actions` parameter are also wrong, for example, for the following request:

```
curl -X POST 'http://localhost/portal/o/headless-delivery/v1.0/sites/20121/blog-postings' \
-H 'accept: application/json' \
-H 'Content-Type: application/json' \
-u 'test@liferay.com:test' \
-d '{
    "headline": "Test Blog Entry headline",
    "articleBody": "Test Blog Entry article body"
}'
```

the response contains incorrect URLs:

```
{
  "actions" : {
    "get-rendered-content-by-display-page" : {
      "method" : "GET",
      "href" : "http://localhost/o/headless-delivery/v1.0/blog-postings/44908/rendered-content-by-display-page/{displayPageKey}"
    },
    "get" : {
      "method" : "GET",
      "href" : "http://localhost/o/headless-delivery/v1.0/blog-postings/44908"
    },
    "replace" : {
      "method" : "PUT",
      "href" : "http://localhost/o/headless-delivery/v1.0/blog-postings/44908"
    },
    "update" : {
      "method" : "PATCH",
      "href" : "http://localhost/o/headless-delivery/v1.0/blog-postings/44908"
    },
    "delete" : {
      "method" : "DELETE",
      "href" : "http://localhost/o/headless-delivery/v1.0/blog-postings/44908"
    }
  },
  [...]
}
```

### **After the PR:**

The OpenAPI is shown:

image

with the proper URL in the `Servers` combo.

For the same request:

```
curl -X POST 'http://localhost/portal/o/headless-delivery/v1.0/sites/20121/blog-postings' \
-H 'accept: application/json' \
-H 'Content-Type: application/json' \
-u 'test@liferay.com:test' \
-d '{
    "headline": "Test Blog Entry headline",
    "articleBody": "Test Blog Entry article body"
}'
```

the URLs are correct:

```
{
  "actions" : {
    "get-rendered-content-by-display-page" : {
      "method" : "GET",
      "href" : "http://localhost/portal/o/headless-delivery/v1.0/blog-postings/45001/rendered-content-by-display-page/{displayPageKey}"
    },
    "get" : {
      "method" : "GET",
      "href" : "http://localhost/portal/o/headless-delivery/v1.0/blog-postings/45001"
    },
    "replace" : {
      "method" : "PUT",
      "href" : "http://localhost/portal/o/headless-delivery/v1.0/blog-postings/45001"
    },
    "update" : {
      "method" : "PATCH",
      "href" : "http://localhost/portal/o/headless-delivery/v1.0/blog-postings/45001"
    },
    "delete" : {
      "method" : "DELETE",
      "href" : "http://localhost/portal/o/headless-delivery/v1.0/blog-postings/45001"
    }
  },
  [...]
}
```

### CODE REVIEW

File: modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/util/UriInfoUtil.java [hunk 1]
```diff
@@ -46,13 +49,15 @@ public static String getBasePath(UriInfo uriInfo) {
 	}
 
 	public static UriBuilder getBaseUriBuilder(UriInfo uriInfo) {
```
Comment (4lejandrito) [inline modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/util/UriInfoUtil.java:51]: I see `getBaseUriBuilder` and `getAbsolutePath` are almost identical. Would it make sense to reuse the logic with a private method that receives the `uriBuilder`? This way if we have to fix something else again in the future we won't have to do it twice. What do you think?

File: modules/apps/portal-vulcan/portal-vulcan-api/src/test/java/com/liferay/portal/vulcan/util/UriInfoUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.mockito.Mockito;
+
+/**
+ * @author Carlos Correa
+ */
+public class UriInfoUtilTest {
```
Comment (LuismiBarcos) [inline modules/apps/portal-vulcan/portal-vulcan-api/src/test/java/com/liferay/portal/vulcan/util/UriInfoUtilTest.java:43]: Thanks for this test class 👏 👏 👏

**Review by LuismiBarcos — APPROVED** [review]
LGTM :)

File: modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/util/UriInfoUtil.java (hunk 1 above)
Comment (ccorreagg) [reply modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/util/UriInfoUtil.java:51]: You are right. Updated!

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Review by 4lejandrito — APPROVED** [review]

**Comment by ccorreagg** [comment]
ci:stop

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:forward

**Pull request closed without merging**